scipy/conftest.py

```diff
@@ -44,3 +46,16 @@ def check_fpu_mode(request):
         warnings.warn("FPU mode changed from {0:#x} to {1:#x} during "
                       "the test".format(old_mode, new_mode),
                       category=FPUModeChangeWarning, stacklevel=0)
+
+
+@pytest.yield_fixture(scope="module")
```

Comment:
Didn't get how this worked, and when looking it up it seems `yield_fixture` is deprecated: https://docs.pytest.org/en/latest/yieldfixture.html for new code. So can you update and/or explain?

Reply:
Yes I can probably redo it. The only purpose is to have it call `gc.collect()` at the end of some test runs. Now that it's only decorating a couple, I could just explicitly do it, though it would require more imports and some explicit `del` statements.

The reason to have it call `gc.collect` at the end of the tests is so that the hang occurs in closer proximity to the problem (namely, the tests that use `Pool` without a context manager) rather than elsewhere (in a `sparse` test much later that uses `gc`).